ember-headless-form/src/components/-private/control/textarea.ts

```diff
@@ -0,0 +1,18 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+export interface HeadlessFormControlTextareaComponentSignature {
+  Element: HTMLTextAreaElement;
+  Args: {
+    value: string;
+    fieldId: string;
+    setValue: (value: string) => void;
+  };
+}
+
+export default class HeadlessFormControlTextareaComponent extends Component<HeadlessFormControlTextareaComponentSignature> {
+  @action
+  handleInput(e: Event | InputEvent): void {
+    this.args.setValue((e.target as HTMLTextAreaElement).value);
```

Comment:
Suggestion, nit: again instead of a cast it would be nice to use a custom type guard here...I think one benefit is folks could re-use the custom type guard instead of always reaching for a cast

Reply:
As in the same case in the other PR, I would add an `assert()` here and remove the type cast, even if that is a bit more verbose than a reusable util function. But `assert()` gets stripped from a production build. And given that the only case where `e.target` is not a `<textarea>` is when _we_ mess this up (by applying `{{on}}` on the wrong element) I would not want to add a runtime overhead by calling a function that is not stripped away...